### PR TITLE
Add plugin system to STS for more hackiness

### PR DESCRIPTION
**Pull request by Zaph-x**
This pull request adds a plugin system to STS, such that the user can customise their sts to their liking

Customisation includes:
- Adding new shorthands
- Adding new commands
- Add new functionality

Commits in pr:
- Added template plugins
- Added plugin system
- Updated gitignore

### CODE REVIEW

**Comment by SEbbaDK** [comment]
Would it be worth having a buy-hook? I have a plugin in mind that would create a custom ware, and then execute something if that ware is attempted bought, but as far as i can see, that isn't possible without hacking around the current system a lot, or am i just missing something?

**Comment by Zaph-x** [comment]
> Would it be worth having a buy-hook? I have a plugin in mind that would create a custom ware, and then execute something if that ware is attempted bought, but as far as i can see, that isn't possible without hacking around the current system a lot, or am i just missing something?
> 

I had this exact thought earlier today, with a pre- and post-sale hook. Definitely worth looking into

File: main.py [hunk 1]
```diff
@@ -472,11 +477,52 @@ def update_script():
...
+
+    if args.user is None:
+        args.user = get_saved_user()
+
+    if not args.noplugins:
+        for plugin in plugins:
+            try:
+                plugin.run(wares, args, arg_array, parser, SHORTHANDS)
+            except:
+                pass
```
Comment (falkecarlsen) [inline main.py:525]: Could be nice to emit exceptions to user when running third-party plugins. Maybe print caught exception?

File: main.py [hunk 2]
```diff
@@ -545,6 +590,13 @@ def main():
         else:
             print_no_user_help(args.user)
 
+    if not args.noplugins:
+        for plugin in plugins:
+            try:
+                plugin.run(wares, args, arg_array, parser, SHORTHANDS)
+            except:
+                pass
+
```
Comment (falkecarlsen) [inline main.py:599]: Duplicate?

**Review by falkecarlsen — CHANGES_REQUESTED** [review]
This change is cursed and I pity future FITministers --- let's do it!

File: main.py (hunk 2 above)
Comment (Zaph-x) [reply main.py]: Resolved in abe2943

File: main.py (hunk 1 above)
Comment (Zaph-x) [reply main.py:549]: Resolved in abe2943

**Review by falkecarlsen — APPROVED** [review]
:shipit:

**Pull request merged**